ios/browser/api/password/brave_password_api.mm

```diff
@@ -0,0 +1,472 @@
+
+    if (dateCreated) {
+      [self setDateCreated:dateCreated];
+    }
+
+    if (dateLastUsed) {
+      [self setDateLastUsed:dateLastUsed];
+    }
+
+    if (datePasswordChanged) {
+      [self setDatePasswordChanged:datePasswordChanged];
+    }
+    if (usernameElement) {
+      [self setUsernameElement:usernameElement];
+    }
+
+    if (usernameValue) {
+      [self setUsernameValue:usernameValue];
+    }
+
+    if (passwordElement) {
+      [self setPasswordElement:passwordElement];
+    }
+
+    if (passwordValue) {
+      [self setPasswordValue:passwordValue];
+    }
+
+    self.isBlockedByUser = isBlockedByUser;
+
+    password_form_scheme_ =
+        brave::ios::PasswordFormSchemeForPasswordFormDigest(scheme);
+  }
+
+  return self;
+}
+
+- (id)copyWithZone:(NSZone*)zone {
+  IOSPasswordForm* passwordFormCopy = [[[self class] allocWithZone:zone] init];
+
+  if (passwordFormCopy) {
+    passwordFormCopy.url = self.url;
+    passwordFormCopy.signOnRealm = self.signOnRealm;
+    passwordFormCopy.dateCreated = self.dateCreated;
+    passwordFormCopy.dateLastUsed = self.dateLastUsed;
+    passwordFormCopy.datePasswordChanged = self.datePasswordChanged;
+    passwordFormCopy.usernameElement = self.usernameElement;
+    passwordFormCopy.usernameValue = self.usernameValue;
+    passwordFormCopy.passwordElement = self.passwordElement;
+    passwordFormCopy.passwordValue = self.passwordValue;
+    passwordFormCopy.isBlockedByUser = self.isBlockedByUser;
+    passwordFormCopy.scheme = self.scheme;
+  }
+
+  return passwordFormCopy;
+}
+
+- (void)updatePasswordForm:(NSString*)usernameValue
+             passwordValue:(NSString*)passwordValue {
+  if ([usernameValue length] != 0) {
+    [self setUsernameValue:usernameValue];
+  }
+
+  if ([passwordValue length] != 0) {
+    [self setPasswordValue:passwordValue];
+  }
+}
+
+- (void)setUrl:(NSURL*)url {
+  gurl_ = net::GURLWithNSURL(url);
+}
+
+- (NSURL*)url {
+  return net::NSURLWithGURL(gurl_);
+}
+
+- (void)setSignOnRealm:(NSString*)signOnRealm {
+  if ([signOnRealm length] != 0) {
+    signon_realm_ = base::SysNSStringToUTF8(signOnRealm);
+  } else {
+    signon_realm_ = gurl_.spec();
+  }
+}
+
+- (NSString*)signOnRealm {
+  return base::SysUTF8ToNSString(signon_realm_);
+}
+
+- (void)setDateCreated:(NSDate*)dateCreated {
+  date_created_ = base::Time::FromNSDate(dateCreated);
+}
+
+- (void)setDateLastUsed:(NSDate*)dateLastUsed {
+  date_last_used_ = base::Time::FromNSDate(dateLastUsed);
+}
+
+- (void)setDatePasswordChanged:(NSDate*)datePasswordChanged {
+  date_created_ = base::Time::FromNSDate(datePasswordChanged);
+}
+
+- (NSDate*)dateCreated {
+  return date_created_.ToNSDate();
+}
+
+- (NSDate*)dateLastUsed {
+  return date_last_used_.ToNSDate();
+}
+
+- (NSDate*)datePasswordChanged {
+  return date_password_changed_.ToNSDate();
+}
+
+- (void)setUsernameElement:(NSString*)usernameElement {
+  username_element_ = base::SysNSStringToUTF16(usernameElement);
```

Comment:
I think everything is fine with passing `nil` into `SysNSStringToUTF16`/`SysNSStringToUTF8`, but you may need to double check.